src/utils/horizontal-bar-transformer.utils.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  ChartSegment,
+  HorizontalBarChartBounds,
+} from '@custom-types/chart.types';
+import { WeightedBounds } from '@custom-types/threshold.types';
+
+export const convertToHorizontalBarChartBounds = <
+  TClassificationLevel extends string
+>(
+  bounds: WeightedBounds<TClassificationLevel>
+): HorizontalBarChartBounds<TClassificationLevel> => ({
+  lowerBound: bounds.lower,
+  upperBounds: Object.keys(bounds.upper).reduce((acc, key) => {
+    acc[key as TClassificationLevel] =
+      bounds.upper[key as TClassificationLevel].value;
+
+    return acc;
+  }, {} as Record<TClassificationLevel, number>),
+});
+
+export const convertToChartSegments = <TClassificationLevel extends string>(
+  bounds: WeightedBounds<TClassificationLevel>
+): ChartSegment[] => {
+  const segments: ChartSegment[] = [];
+
+  let currentStart = bounds.lower;
+
+  Object.keys(bounds.upper).forEach((classification) => {
```

Comment:
I don't think we need Object.keys + forEach here - you're using keys to get the keys in an array, forEach to loop over that array and then getting the classificationBound inside each iteration... building the segment for each one - could use Object.entries instead to reduce iterations required

Reply:
This should allow you to remove the `as TClassificationLevel` casting too

Reply:
same again for function above

Reply:
after discussion could just use Object.values and remove cast

Reply:
resolved. as object.values removes the typing i added a type safe function to get the values back typed.